plugin/src/main/java/io/jenkins/plugins/remotingkafka/GlobalKafkaConfiguration.java

```diff
@@ -1,17 +1,43 @@
 package io.jenkins.plugins.remotingkafka;
 
+import com.cloudbees.plugins.credentials.CredentialsMatchers;
+import com.cloudbees.plugins.credentials.CredentialsProvider;
+import com.cloudbees.plugins.credentials.common.StandardListBoxModel;
+import com.cloudbees.plugins.credentials.common.StandardUsernamePasswordCredentials;
+import com.cloudbees.plugins.credentials.domains.SchemeRequirement;
+import com.cloudbees.plugins.credentials.matchers.IdMatcher;
 import hudson.Extension;
+import hudson.model.Item;
+import hudson.security.ACL;
 import hudson.util.FormValidation;
+import hudson.util.ListBoxModel;
 import jenkins.model.GlobalConfiguration;
+import jenkins.model.Jenkins;
 import net.sf.json.JSONObject;
 import org.apache.commons.lang.StringUtils;
+import org.kohsuke.stapler.AncestorInPath;
 import org.kohsuke.stapler.QueryParameter;
 import org.kohsuke.stapler.StaplerRequest;
 
+import javax.servlet.ServletException;
+import java.io.IOException;
+import java.net.Socket;
+import java.util.Collections;
+import java.util.List;
+
 @Extension
 public class GlobalKafkaConfiguration extends GlobalConfiguration {
-    private String connectionURL;
+    public static final SchemeRequirement KAFKA_SCHEME = new SchemeRequirement("kafka");
+    private String credentialsId;
+    private String brokerURL;
     private String zookeeperURL;
+    private String username;
+    private String password;
```

Comment:
🐛 In such case...
* passwords will be persisted in plain text if somebody saves the configuration
* passwords/username will be outdated if the credential gets reconfigured

I would propose to always retrieve data from credentials on-demand